x/stake/queryable.go

```diff
@@ -0,0 +1,234 @@
+	QueryDelegatorValidator  = "delegatorValidator"
+	QueryPool                = "pool"
+	QueryParameters          = "parameters"
+)
+
+// creates a querier for staking REST endpoints
+func NewQuerier(k keep.Keeper, cdc *wire.Codec) sdk.Querier {
+	return func(ctx sdk.Context, path []string, req abci.RequestQuery) (res []byte, err sdk.Error) {
+		switch path[0] {
+		case QueryValidators:
+			return queryValidators(ctx, cdc, k)
+		case QueryValidator:
+			return queryValidator(ctx, cdc, req, k)
+		case QueryDelegator:
+			return queryDelegator(ctx, cdc, req, k)
+		case QueryDelegation:
+			return queryDelegation(ctx, cdc, req, k)
+		case QueryUnbondingDelegation:
+			return queryUnbondingDelegation(ctx, cdc, req, k)
+		case QueryDelegatorValidators:
+			return queryDelegatorValidators(ctx, cdc, req, k)
+		case QueryDelegatorValidator:
+			return queryDelegatorValidator(ctx, cdc, req, k)
+		case QueryPool:
+			return queryPool(ctx, cdc, k)
+		case QueryParameters:
+			return queryParameters(ctx, cdc, k)
+		default:
+			return nil, sdk.ErrUnknownRequest("unknown stake query endpoint")
+		}
+	}
+}
+
+// defines the params for the following queries:
+// - 'custom/stake/delegator'
+// - 'custom/stake/delegatorValidators'
+type QueryDelegatorParams struct {
+	DelegatorAddr sdk.AccAddress
+}
+
+// defines the params for the following queries:
+// - 'custom/stake/validator'
+type QueryValidatorParams struct {
+	ValidatorAddr sdk.ValAddress
+}
+
+// defines the params for the following queries:
+// - 'custom/stake/delegation'
+// - 'custom/stake/unbondingDelegation'
+// - 'custom/stake/delegatorValidator'
+type QueryBondsParams struct {
+	DelegatorAddr sdk.AccAddress
+	ValidatorAddr sdk.ValAddress
+}
+
+func queryValidators(ctx sdk.Context, cdc *wire.Codec, k keep.Keeper) (res []byte, err sdk.Error) {
+	validators := k.GetBechValidators(ctx)
+
+	res, errRes := wire.MarshalJSONIndent(cdc, validators)
+	if err != nil {
+		return nil, sdk.ErrInternal(fmt.Sprintf("could not marshal result to JSON: %s", errRes.Error()))
+	}
+	return res, nil
+}
+
+func queryValidator(ctx sdk.Context, cdc *wire.Codec, req abci.RequestQuery, k keep.Keeper) (res []byte, err sdk.Error) {
+	var params QueryValidatorParams
+
+	errRes := cdc.UnmarshalJSON(req.Data, &params)
+	if errRes != nil {
+		return []byte{}, sdk.ErrUnknownAddress(fmt.Sprintf("incorrectly formatted request address: %s", err.Error()))
+	}
+
+	validator, found := k.GetValidator(ctx, params.ValidatorAddr)
+	if !found {
+		return []byte{}, ErrNoValidatorFound(DefaultCodespace)
+	}
+
+	bechValidator, errRes := validator.Bech32Validator()
+	if err != nil {
+		return nil, sdk.ErrInternal(fmt.Sprintf("could not bech32ify validator: %s", errRes.Error()))
+	}
+
+	res, errRes = wire.MarshalJSONIndent(cdc, bechValidator)
+	if errRes != nil {
+		return nil, sdk.ErrInternal(fmt.Sprintf("could not marshal result to JSON: %s", errRes.Error()))
+	}
+	return res, nil
+}
+
+// TODO query with limit
```

Comment:
If we're going to put TODOs like these, we should also include an issue reference 👍